docs/docs/dev_docs/contracts/syntax/slow_updates_tree.md

```diff
@@ -0,0 +1,268 @@
+
+## SlowTree.nr Contract
+
+This is a smart contract developed by Aztec that establishes and manages a slow updates tree structure. It allows developers to access and interact with the tree, such as reading and updating data.
+
+You can find it [here](https://github.com/AztecProtocol/aztec-packages/tree/master/yarn-project/noir-contracts/src/contracts/slow_tree_contract).
+
+## SlowMap type
+
+This is a type in the Aztec.nr library that is utilized by the SlowTree contract. It defines the underlying data structure for the slow updates tree, and handles storing both the current and pending values for each data entry. 
+
+You can find it [here](https://github.com/AztecProtocol/aztec-nr/blob/master/slow-updates-tree/src/slow_map.nr).
+
+The diagram below describes how these components work together. It does not contain all the functionality. 
+
+```mermaid
+graph TD
+    MSC[Main Smart Contract] --> INT[Interface]
+    STC --> SMT
+    
+    INT_RAP[read_at_pub] <--> STC_RAP[read_at_public]
+    INT_RA[read_at] <--> STC_RA[read_at]
+    INT_UAP[update_at_public] <--> STC_UAP[update_at_public]
+    INT_UA[update_at_private] <--> STC_UA[update_at_private]
+
+    STC_RA <--> VMP[verify_membership_proof]
+    STC_UA <--> CR[compute_roots]
+    
+    subgraph INT[Interface]
+        INT_RAP
+        INT_UAP
+        INT_RA
+        INT_UA
+    end
+    
+    subgraph STC[SlowTree.nr]
+        STC_RAP
+        STC_UAP
+        STC_RA
+        STC_UA
+    end
+    
+    subgraph SMT[SlowMap Type]
+        Change{Epoch Over} -->|True| Current{Current}
+        Change -->|False| Pending{Pending}
+        Current --> Current1[Current Commitment 1]
+        Current --> CurrentM[Current Commitment M]
+        CurrentM --> Value1[Current Value 1]
+        CurrentM --> Value2[Current Value 2]
+        CurrentM --> ValueN[Current Value N]
+        Pending --> PendingM[Pending Commitment 1]
+        PendingM --> PValue1[Pending Value 1]
+        PendingM --> PValue2[Pending Value 2]
+        PendingM --> PValueN[Pending Value N]
+    end
+
+    style INT fill:#fff,stroke:#333,stroke-width:1px
+    style STC fill:#fff,stroke:#333,stroke-width:1px
+    style SMT fill:#fff,stroke:#333,stroke-width:1px
+```
+
+# How to integrate a slow updates tree
+
+1. Copy the *SlowTree.nr* example and its dependencies, found [here](https://github.com/AztecProtocol/aztec-packages/tree/master/yarn-project/noir-contracts/src/contracts/slow_tree_contract). Replace the constants with whatever you like and deploy it to your sandbox
+2. Copy the *SlowMap interface* for easy interaction with your deployed SlowTree. Find it [here](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/interfaces.nr)
```

Comment:
we are sometimes hyperlinking code and sometimes #include_code. Should we aim for consistency?

Reply:
hyperlinking is for the entire file to copy & paste. do you think i should replace it with include_code? its very long